internal/pkg/service/stream/storage/node/coordinator/fileimport/operator.go

```diff
@@ -214,11 +245,17 @@ func (o *operator) checkFile(ctx context.Context, file *fileData) {
 		return
 	}
 
-	if !file.Retry.Allowed(o.clock.Now()) {
+	if !file.Retryable.Allowed(o.clock.Now()) {
+		return
+	}
+
+	// Skip file import if sink is deleted or disabled
+	sink, ok := o.sinks.Get(file.FileKey.SinkKey)
+	if !ok || !sink.Enabled {
```

Comment:
Just to be sure, wouldn't it be better to `incrementRetryTimeout` if the sink is no enabled or not existing. Doing such check when sink is disabled forever utilizes in my opinion pointlessly the CPU/MEM

Reply:
We can call about it if you want.

Reply:
I understand that the intent of `IncrementRetryAttempt` was originally to be called when `etcd` change of state is not performed (e.g entity is not changed or we could not increase the attempt which is also `etcd` operation). This is repository function that communicates with `etcd`. I would expect that this would increase the timeout **but leave entity as not processed** to load off the loops.

We can leave it as it is and make another set of benchmarks. If it is a problem we will do appropriate adjustements.